src/main/java/org/sunbird/hubservices/daoimpl/GraphDao.java

```diff
@@ -97,33 +98,63 @@ public Boolean upsertRelation(Node nodeFrom, Node nodeTo, Map<String, String> re
+                    transaction.commitAsync().toCompletableFuture().get();
+                    logger.info("user relation with toUUID {} and fromUUID {} updated successfully ", nodeTo.getId(),
+                            nodeFrom.getId());
                 }
-
-                String query = "MATCH (n:" + label + ")-[r:connect]->(n1:" + label +
-                        ") WHERE n.id = $fromUUID AND n1.id = $toUUID " + "SET r" + " += " +
-                        "$props " + "RETURN n,n1";
-
-                statement = new Statement(query, parameters);
-                result = transaction.run(statement);
-                recordSize = result.list().size();
-                result.consume();
-                if (recordSize == 0) { // nodes relation doesn't exists
-                    createRelationshipBetweenTwoNodes(nodeFrom, nodeTo, transaction, parameters);
+                else {
+                    String query = "MATCH (n:" + label + ")-[r:connect]->(n1:" + label +
+                            ") WHERE n.id = $fromUUID AND n1.id = $toUUID " + "RETURN n,n1";
+
+                    statement = new Statement(query, parameters);
+                    result = transaction.run(statement);
+                    recordSize = result.list().size();
+                    result.consume();
+                    if (recordSize == 0) { // nodes relation doesn't exists
+                        createRelationshipBetweenTwoNodes(nodeFrom, nodeTo, transaction, parameters);
+                    }
+                    else
+                    {
+                        updateRelationshipBetweenTwoNodes("Nodes exists, Updating Relationship Properties!", nodeTo, nodeFrom, statement, result, transaction, recordSize, relationProperties);
+                    }
+                    transaction.commitAsync().toCompletableFuture().get();
+                    logger.info("user relation with toUUID {} and fromUUID {} added successfully ", nodeTo.getId(),
+                            nodeFrom.getId());
                 }
-                transaction.commitAsync().toCompletableFuture().get();
-                logger.info("user relation with toUUID {} and fromUUID {} updated successfully ", nodeTo.getId(),
-                        nodeFrom.getId());
-
-            } catch (ClientException e) {
-                throw new GraphException(ErrorCode.GRAPH_TRANSACTIONAL_ERROR.name(), e.getMessage());
+            } catch (Exception e) {
+                transaction.rollbackAsync().toCompletableFuture().get();
+                logger.error("user relation creation failed : ", e);
+                return Boolean.FALSE;
 
+            } finally {
+                try {
+                    if (session != null && transaction != null) {
+                        transaction.close();
+                        session.close();
+                    }
+                } catch (Exception e) {
+                    logger.error("Session Closed : ", e);
+                }
```

Comment:
By using try with resources --- objects are closed automatically -- since it's implements AutoCloseable... no need to write this again...